File: 822-unique-morse-code-words/unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        vector<string> morse = {
        ".-","-...","-.-.","-..",".","..-.","--.","....","..",
        ".---","-.-",".-..","--","-.","---",".--.","--.-",".-.",
        "...","-","..-","...-",".--","-..-","-.--","--.."
    };
    
    unordered_set<string> unique;
    
    for (const string& word : words) {
        string code = "";
        for (char c : word) {
            code += morse[c - 'a'];
        }
        unique.insert(code);
    }
    
    return unique.size();

    }
};

File: 822-unique-morse-code-words/unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        morse_alpha = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]

        seen = set() # stores the transformation seen so far
        for word in words:
            morse = ""
            for char in word:
                idx = ord(char)-ord("a")
                morse+=morse_alpha[idx]
            seen.add(morse)

        return len(seen)
